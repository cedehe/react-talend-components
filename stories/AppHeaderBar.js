import React from 'react';
import { storiesOf, action } from '@kadira/storybook';

import { AppHeaderBar } from '../src';

/*************************************** TYPEAHEAD ************************************************/
const exampleId = 'component-id';
const focusedItemIndex = 1;
const focusedSectionIndex = 0;
const value = 'le';
const renderItemData = { value };
const inputProps = {
	value,
	placeholder: 'Search anything',
};
const itemProps = {};
const typeaheadItems = [
	{
		title: 'category 1',
		icon: {
			name: 'fa fa-filter',
			title: 'icon'
		},
		suggestions: [
			{
				title: 'le title 1',
				description: 'description: Uxoresque est in pacto est marito est hastam nomine in eos discessura incredibile tempus ardore.',
			},
			{
				title: 'title 2 les elephants elementaires ont des aile ',
				description: 'description: Aut aut cum satis inter Epicuri quidem cum erat inquam controversia autem mihi utrumque Attico.',
			},
		],
	},

	{
		title: 'category 2',
		icon: {
			name: 'fa fa-asterisk',
			title: 'icon',
		},
		suggestions: [
			{
				title: 'title 3',
				description: 'description: In sanciatur libere audeamus exspectemus amicitia et dum ne audeamus causa monendum honesta studium valeat.',
			},
		],
	},

	{
		title: 'category 3',
		icon: {
			name: 'fa fa-asterisk',
			title: 'icon',
		},
		suggestions: [
			{
				title: 'title 4',
				description: 'description: Praesentibus genero ne in Africani mandavi saepius ipsam C in libro et hoc Laeli cum.',
			},
			{
				title: 'title 5',
				description: 'description: Feceris unde tot illo tot clientes dederis numerando et indiscretus cum paria et unde ubi.',
			},
			{
				title: 'title 6',
				description: 'description: Gradu quos cedentium sunt appeterent ita ancoralia instar luna sunt etiam ubi incendente nihil observabant.',
			},
		],
	}];

const search = {
	id: exampleId,
	config: {
		icon: {
			name: 'fa fa-search',
			title: 'icon',
		},
	},
	items: typeaheadItems,
	inputProps,
	itemProps,
	focusedItemIndex,
	focusedSectionIndex,
	renderItemData,
};

/****************************************** FORM **************************************************/
const inputForm = {
	forms: [{
		form: {
			pullRight: true,
			onSubmit: action('onSubmit'),
			role: 'search',
		},
		formgroups: [
			{
				formgroup: {
					controlId: 'globalSearch',
					pullRight: true,
				},
				formcontrol: {
					type: 'text',
					placeholder: 'search anything',
					onChange: action('onChange in search'),
				},
			},
		],
		button: {
			onClick: action('onClick form'),
			bsStyle: 'link',
		},
		buttonLabel: 'submit',
	}],
};

/************************************** AppHeaderBar Props ****************************************/
const props = {
	app: 'Example App Name',
	brandLink: {
		onClick: action('brandonClick'),
	},
	content: [
		{
			navs: [
				{
					navItems: [
						{
							type: 'navItem',
							item: {
								icon: 'fa fa-bars',
								name: 'hello',
								onClick: action('onClick bars')
							},
						},
					],
				},
				{
					nav: { pullRight: true },
					navItems: [
						{
							type: 'navItem',
							item: {
								icon: 'fa fa-heart',
								name: 'world',
								onClick: action('onClick heart')
							},
						},
						{
							type: 'dropdown',
							item: {
								dropdown: {
									id: 'myDropdown',
									title: 'user 1',
									onSelect: action('dropdown onSelect'),
								},
								items: [
									{
										icon: 'fa fa-fw fa-cog',
										name: 'settings',
										onClick: action('onClick settings')
									},
								],
							},
						},
					],
				},
			],
		},
	],
};

storiesOf('App Header Bar', module)
	.addWithInfo('default', () => {
		props.content[1] && delete props.content[1];
		return (
		<div>
			<AppHeaderBar {...props} />
			<div className="container" style={{ paddingTop: 40 }}>
				<h1>AppHeaderBar</h1>
				<h2>Definition</h2>
				<p>Display a navigation bar on top of the page.</p>
				<h2>Examples</h2>
				<p>Look on top</p>
				<pre>{JSON.stringify(props, null, 2)}</pre>
			</div>
		</div>);
	})

	.addWithInfo('Without brandLink', () => {
		const myprops = Object.assign({}, props);
		delete myprops.brandLink;
		myprops.content[1] && delete myprops.content[1];
		return (
			<div>
				<AppHeaderBar {...myprops} />
				<div className="container" style={{ paddingTop: 40 }}>
					<h1>AppHeaderBar</h1>
					<h2>Definition</h2>
					<p>Display a navigation bar on top of the page.</p>
					<p>Look on top</p>
				</div>
			</div>
		);
	})

	.addWithInfo('With form', () => {
		props.content[1] = inputForm;
		return (
			<div>
				<AppHeaderBar {...props} />
				<div className="container" style={{ paddingTop: 40 }}>
					<h1>AppHeaderBar</h1>
					<h2>Definition</h2>
					<p>Display a navigation bar on top of the page.</p>
					<p>Look on top</p>
				</div>
			</div>
		);
	})

	.addWithInfo('With Typeahead Icon', () => {
		props.content[1] =
			{
				search: {
					id: exampleId,
					config: {
						isOnlyIcon: true,
						icon: {
							name: 'fa fa-search',
							title: 'icon',
							actionStyle: 'link',
						},
						onInputIconClick: action('icon clicked'),
					},
				},
			};
		return (
			<div>
				<AppHeaderBar {...props} />
				<div className="container" style={{ paddingTop: 40 }}>
					<h1>AppHeaderBar</h1>
					<h2>Definition</h2>
					<p>Display a navigation bar on top of the page.</p>
					<p>Look on top</p>
				</div>
			</div>
		);
	})

	.addWithInfo('With Typeahead only input', () => {
		props.content[1] = {
			search: {
				id: exampleId,
				config: {
					icon: {
						name: 'fa fa-search',
						title: 'icon',
					},
				},
				inputProps: inputProps,
				itemProps: itemProps,
				focusedItemIndex: focusedItemIndex,
				focusedSectionIndex: focusedSectionIndex,
				renderItemData: renderItemData,
			}
		};
		return (
			<div>
				<AppHeaderBar {...props} />
				<div className="container" style={{ paddingTop: 40 }}>
					<h1>AppHeaderBar</h1>
					<h2>Definition</h2>
					<p>Display a navigation bar on top of the page.</p>
					<p>Look on top</p>
				</div>
			</div>
		);
	})

	.addWithInfo('With Typeahead items', () => {
		props.content[1] = { search };
		return (
			<div>
				<AppHeaderBar {...props} />
				<div className="container" style={{ paddingTop: 40 }}>
					<h1>AppHeaderBar</h1>
					<h2>Definition</h2>
					<p>Display a navigation bar on top of the page.</p>
					<p>Look on top</p>
				</div>
			</div>
		);
	})
;
