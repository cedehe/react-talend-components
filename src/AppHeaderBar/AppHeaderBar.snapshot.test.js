import React from 'react';
import renderer from 'react-test-renderer';
import AppHeaderBar from './AppHeaderBar.component';
import {
	exampleId,
	focusedItemIndex,
	focusedSectionIndex,
	renderItemData,
	inputProps,
	itemProps,
	search,
	inputForm,
	props,
} from './test.props';

jest.mock('react-dom');

describe('AppHeaderBar', () => {
	it('should render', () => {
		// given
		const appBar = <AppHeaderBar {...props} />;

		// when
		const appBarInstance = renderer.create(appBar).toJSON();

		// then
		expect(appBarInstance).toMatchSnapshot();
	});

	it('should render without brandLink', () => {
		// given
		const myprops = Object.assign({}, props);
		delete myprops.brandLink;
		const appBar = <AppHeaderBar {...myprops} />;

		// when
		const appBarInstance = renderer.create(appBar).toJSON();

		// then
		expect(appBarInstance).toMatchSnapshot();
	});

	it('should render with form', () => {
		// given
		props.content[1] = inputForm;
		const appBar = <AppHeaderBar {...props} />;

		// when
		const appBarInstance = renderer.create(appBar).toJSON();

		// then
		expect(appBarInstance).toMatchSnapshot();
	});

	it('should render with Typeahead Icon', () => {
		// given
		props.content[1] =
		{
			search: {
				id: exampleId,
				config: {
					isOnlyIcon: true,
					icon: {
						name: 'fa fa-search',
						title: 'icon',
						actionStyle: 'link',
					},
					onInputIconClick: jest.fn(),
				},
			},
		};
		const appBar = <AppHeaderBar {...props} />;

		// when
		const appBarInstance = renderer.create(appBar).toJSON();

		// then
		expect(appBarInstance).toMatchSnapshot();
	});

	it('should render with only Typeahead input', () => {
		// given
		props.content[1] = {
			search: {
				id: exampleId,
				config: {
					icon: {
						name: 'fa fa-search',
						title: 'icon',
					},
				},
				inputProps,
				itemProps,
				focusedItemIndex,
				focusedSectionIndex,
				renderItemData,
			},
		};
		const appBar = <AppHeaderBar {...props} />;

		// when
		const appBarInstance = renderer.create(appBar).toJSON();

		// then
		expect(appBarInstance).toMatchSnapshot();
	});

	it('should render with Typeahead items', () => {
		// given
		props.content[1] = { search };
		const appBar = <AppHeaderBar {...props} />;

		// when
		const appBarInstance = renderer.create(appBar).toJSON();

		// then
		expect(appBarInstance).toMatchSnapshot();
	});
});
