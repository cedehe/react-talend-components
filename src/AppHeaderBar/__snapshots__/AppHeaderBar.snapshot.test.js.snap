exports[`AppHeaderBar should render 1`] = `
<nav
  className="tc-app-header-bar navbar navbar-inverse navbar-fixed-top">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <button
        className="my-brand-class navbar-brand btn btn-link"
        disabled={false}
        onClick={[Function]}
        role="link"
        title="my brand title"
        type="button">
        Example App Name
      </button>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse">
      <ul
        className="nav navbar-nav"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="hello"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-bars"
              title={null} />
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="world"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-heart"
              title={null} />
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href=""
            id="myDropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            user 1
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="myDropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href=""
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                <i
                  aria-hidden="true"
                  className="fa fa-fw fa-cog"
                  title={null} />
                settings
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`AppHeaderBar should render with Typeahead Icon 1`] = `
<nav
  className="tc-app-header-bar navbar navbar-inverse navbar-fixed-top">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <button
        className="my-brand-class navbar-brand btn btn-link"
        disabled={false}
        onClick={[Function]}
        role="link"
        title="my brand title"
        type="button">
        Example App Name
      </button>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse">
      <ul
        className="nav navbar-nav"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="hello"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-bars"
              title={null} />
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="world"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-heart"
              title={null} />
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href=""
            id="myDropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            user 1
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="myDropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href=""
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                <i
                  aria-hidden="true"
                  className="fa fa-fw fa-cog"
                  title={null} />
                settings
              </a>
            </li>
          </ul>
        </li>
      </ul>
      <div
        className="navbar-form navbar-right"
        role="search">
        <button
          aria-describedby={42}
          className="btn btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          role={null}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-search"
            title={null} />
        </button>
      </div>
    </div>
  </div>
</nav>
`;

exports[`AppHeaderBar should render with Typeahead items 1`] = `
<nav
  className="tc-app-header-bar navbar navbar-inverse navbar-fixed-top">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <button
        className="my-brand-class navbar-brand btn btn-link"
        disabled={false}
        onClick={[Function]}
        role="link"
        title="my brand title"
        type="button">
        Example App Name
      </button>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse">
      <ul
        className="nav navbar-nav"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="hello"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-bars"
              title={null} />
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="world"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-heart"
              title={null} />
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href=""
            id="myDropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            user 1
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="myDropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href=""
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                <i
                  aria-hidden="true"
                  className="fa fa-fw fa-cog"
                  title={null} />
                settings
              </a>
            </li>
          </ul>
        </li>
      </ul>
      <div
        className="navbar-form navbar-right"
        role="search">
        <div
          style={Object {}}>
          <div
            className={undefined}>
            <label
              className="control-label sr-only"
              htmlFor="react-autowhatever-my-search-input">
              Search
            </label>
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={true}
              aria-haspopup={true}
              aria-owns="react-autowhatever-my-search"
              autoComplete="off"
              autoFocus={true}
              className="form-control"
              id="react-autowhatever-my-search-input"
              onBlur={undefined}
              onChange={undefined}
              onKeyDown={undefined}
              placeholder="Search anything"
              role="combobox"
              style={Object {}}
              type="text"
              value="le" />
            <div
              className={undefined}>
              <i
                aria-hidden="true"
                className="fa fa-search"
                title="icon" />
            </div>
          </div>
          <div
            id="react-autowhatever-my-search"
            style={Object {}}>
            <div
              style={Object {}}>
              <div
                style={Object {}}>
                <div
                  className={undefined}>
                  <i
                    aria-hidden="true"
                    className="fa fa-filter"
                    title="icon" />
                  <span
                    className={undefined}>
                    category 1
                  </span>
                </div>
              </div>
              <ul
                role="listbox"
                style={Object {}}>
                <li
                  id="react-autowhatever-my-search-section-0-item-0"
                  onClick={undefined}
                  role="option"
                  style={Object {}}>
                  <div
                    className={undefined}>
                    <span
                      className={undefined}>
                      <span>
                        
                        <em
                          className={undefined}>
                          le
                        </em>
                      </span>
                      <span>
                         tit
                        <em
                          className={undefined}>
                          le
                        </em>
                      </span>
                      <span>
                         1
                      </span>
                    </span>
                    <p
                      className={undefined}>
                      description: Uxoresque est in pacto est marito est hastam nomine in eos discessura incredibile tempus ardore.
                    </p>
                  </div>
                </li>
                <li
                  id="react-autowhatever-my-search-section-0-item-1"
                  onClick={undefined}
                  role="option"
                  style={Object {}}>
                  <div
                    className={undefined}>
                    <span
                      className={undefined}>
                      <span>
                        tit
                        <em
                          className={undefined}>
                          le
                        </em>
                      </span>
                      <span>
                         2 
                        <em
                          className={undefined}>
                          le
                        </em>
                      </span>
                      <span>
                        s e
                        <em
                          className={undefined}>
                          le
                        </em>
                      </span>
                      <span>
                        phants e
                        <em
                          className={undefined}>
                          le
                        </em>
                      </span>
                      <span>
                        mentaires ont des ai
                        <em
                          className={undefined}>
                          le
                        </em>
                      </span>
                      <span>
                         
                      </span>
                    </span>
                    <p
                      className={undefined}>
                      description: Aut aut cum satis inter Epicuri quidem cum erat inquam controversia autem mihi utrumque Attico.
                    </p>
                  </div>
                </li>
              </ul>
            </div>
            <div
              style={Object {}}>
              <div
                style={Object {}}>
                <div
                  className={undefined}>
                  <i
                    aria-hidden="true"
                    className="fa fa-asterisk"
                    title="icon" />
                  <span
                    className={undefined}>
                    category 2
                  </span>
                </div>
              </div>
              <ul
                role="listbox"
                style={Object {}}>
                <li
                  id="react-autowhatever-my-search-section-1-item-0"
                  onClick={undefined}
                  role="option"
                  style={Object {}}>
                  <div
                    className={undefined}>
                    <span
                      className={undefined}>
                      <span>
                        tit
                        <em
                          className={undefined}>
                          le
                        </em>
                      </span>
                      <span>
                         3
                      </span>
                    </span>
                    <p
                      className={undefined}>
                      description: In sanciatur libere audeamus exspectemus amicitia et dum ne audeamus causa monendum honesta studium valeat.
                    </p>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;

exports[`AppHeaderBar should render with form 1`] = `
<nav
  className="tc-app-header-bar navbar navbar-inverse navbar-fixed-top">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <button
        className="my-brand-class navbar-brand btn btn-link"
        disabled={false}
        onClick={[Function]}
        role="link"
        title="my brand title"
        type="button">
        Example App Name
      </button>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse">
      <ul
        className="nav navbar-nav"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="hello"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-bars"
              title={null} />
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="world"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-heart"
              title={null} />
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href=""
            id="myDropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            user 1
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="myDropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href=""
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                <i
                  aria-hidden="true"
                  className="fa fa-fw fa-cog"
                  title={null} />
                settings
              </a>
            </li>
          </ul>
        </li>
      </ul>
      <div
        className="navbar-form navbar-right"
        onSubmit={[Function]}>
        <div
          className="form-group">
          <input
            className="form-control"
            id="globalSearch"
            onChange={[Function]}
            placeholder="search anything"
            type="text" />
        </div>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button">
          Save
        </button>
      </div>
    </div>
  </div>
</nav>
`;

exports[`AppHeaderBar should render with only Typeahead input 1`] = `
<nav
  className="tc-app-header-bar navbar navbar-inverse navbar-fixed-top">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <button
        className="my-brand-class navbar-brand btn btn-link"
        disabled={false}
        onClick={[Function]}
        role="link"
        title="my brand title"
        type="button">
        Example App Name
      </button>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse">
      <ul
        className="nav navbar-nav"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="hello"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-bars"
              title={null} />
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="world"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-heart"
              title={null} />
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href=""
            id="myDropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            user 1
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="myDropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href=""
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                <i
                  aria-hidden="true"
                  className="fa fa-fw fa-cog"
                  title={null} />
                settings
              </a>
            </li>
          </ul>
        </li>
      </ul>
      <div
        className="navbar-form navbar-right"
        role="search">
        <div
          style={Object {}}>
          <div
            className={undefined}>
            <label
              className="control-label sr-only"
              htmlFor="react-autowhatever-my-search-input">
              Search
            </label>
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-expanded={false}
              aria-haspopup={false}
              aria-owns="react-autowhatever-my-search"
              autoComplete="off"
              autoFocus={true}
              className="form-control"
              id="react-autowhatever-my-search-input"
              onBlur={undefined}
              onChange={undefined}
              onKeyDown={undefined}
              placeholder="Search anything"
              role="combobox"
              style={Object {}}
              type="text"
              value="le" />
          </div>
          <div
            id="react-autowhatever-my-search"
            style={Object {}} />
        </div>
      </div>
    </div>
  </div>
</nav>
`;

exports[`AppHeaderBar should render without brandLink 1`] = `
<nav
  className="tc-app-header-bar navbar navbar-inverse navbar-fixed-top">
  <div
    className="container-fluid">
    <div
      className="navbar-header">
      <span
        className="navbar-brand">
        Example App Name
      </span>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button">
        <span
          className="sr-only">
          Toggle navigation
        </span>
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
        <span
          className="icon-bar" />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse">
      <ul
        className="nav navbar-nav"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="hello"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-bars"
              title={null} />
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-nav navbar-right"
        role={null}>
        <li
          className=""
          role="presentation"
          style={undefined}>
          <a
            href=""
            name="world"
            onClick={[Function]}
            role="button">
            <i
              aria-hidden="true"
              className="fa fa-heart"
              title={null} />
          </a>
        </li>
        <li
          className="dropdown"
          style={undefined}>
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle"
            href=""
            id="myDropdown"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button">
            user 1
             
            <span
              className="caret" />
          </a>
          <ul
            aria-labelledby="myDropdown"
            className="dropdown-menu"
            role="menu">
            <li
              className=""
              role="presentation"
              style={undefined}>
              <a
                href=""
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1">
                <i
                  aria-hidden="true"
                  className="fa fa-fw fa-cog"
                  title={null} />
                settings
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
